specs/agents/transport.md

```diff
@@ -62,7 +62,12 @@ The APM Server accepts both uncompressed and compressed HTTP requests. The follo
 - zlib data format (`Content-Encoding: deflate`)
 - gzip data format (`Content-Encoding: gzip`)
 
-Agents should compress the HTTP payload by default, optimising for speed over compactness (typically known as the "best speed" level).
+Agents MUST compress the HTTP payload, optimising for speed over compactness (typically known as the "best speed" level).
```

Comment:
Would this be better off as a SHOULD instead of a MUST?  I'd be concerned about languages that might not have access to the best speed level, or higher level language where some quirk of their glue code led to unintended perf. issues outside the compression algorithm itself.